docs/guides/verify-shapella-upgrade-manual.md

```diff
@@ -0,0 +1,368 @@
+- `SET_BEACON_REPORT_RECEIVER` `[MN, GL]`
+
+### **AccountingOracle (OZ)**
+*OZ, Proxy*
+
+- PROXY ADMIN
+    - Agent
+- `DEFAULT_ADMIN_ROLE`
+    - Agent
+- `SUBMIT_DATA_ROLE`
+    - None
+- `MANAGE_CONSENSUS_CONTRACT_ROLE`
+    - None
+- `MANAGE_CONSENSUS_VERSION_ROLE`
+    - None
+
+### **Burner**
+*OZ, No Proxy*
+
+- `DEFAULT_ADMIN_ROLE`
+    - Agent
+- `REQUEST_BURN_SHARES_ROLE`
+    - granted to Lido in Burner constructor
+    - NodeOperatorsRegistry
+- `REQUEST_BURN_MY_STETH_ROLE`
+    - None
+- `RECOVER_ASSETS_ROLE`
+    - None
+
+### **LidoLocator (OZ)**
+*OZ, Proxy*
+
+- PROXY ADMIN
+    - Agent
+
+### **StakingRouter (OZ)**
+*OZ, Proxy*
+
+- PROXY ADMIN
+    - Agent
+- `DEFAULT_ADMIN_ROLE`
+    - Agent
+- `MANAGE_WITHDRAWAL_CREDENTIALS_ROLE`
+    - None
+- `STAKING_MODULE_PAUSE_ROLE`
+    - DepositSecurityModule
+- `STAKING_MODULE_RESUME_ROLE`
+    - DepositSecurityModule
+- `STAKING_MODULE_MANAGE_ROLE`
+    - None
+- `REPORT_EXITED_VALIDATORS_ROLE`
+    - AccountingOracle
+- `UNSAFE_SET_EXITED_VALIDATORS_ROLE`
+    - None
+- `REPORT_REWARDS_MINTED_ROLE`
+    - Lido
+
+### **HashConsensus for AccountingOracle (OZ)**
+*OZ, No Proxy*
+
+- `DEFAULT_ADMIN_ROLE`
+    - Agent
+- `MANAGE_MEMBERS_AND_QUORUM_ROLE`
+    - None
+- `DISABLE_CONSENSUS_ROLE`
+    - None
+- `MANAGE_FRAME_CONFIG_ROLE`
+    - None
+- `MANAGE_FAST_LANE_CONFIG_ROLE`
+    - None
+- `MANAGE_REPORT_PROCESSOR_ROLE`
+    - None
+- `address[] _memberAddresses`
+    - Current LidoOracle Committee (migrated in UpgradeTemplate)
+
+### **DepositSecurityModule**
+*Plain Owner, No Proxy*
+
+- owner
+    - Agent
+- guardians
+    - Current DSM guardians committee
+
+### **HashConsensus for ValidatorExitBusOracle (OZ)**
+*OZ, No Proxy*
+
+- `DEFAULT_ADMIN_ROLE`
+    - Agent
+- `MANAGE_MEMBERS_AND_QUORUM_ROLE`
+    - None
+- `DISABLE_CONSENSUS_ROLE`
+    - None
+- `MANAGE_FRAME_CONFIG_ROLE`
+    - None
+- `MANAGE_FAST_LANE_CONFIG_ROLE`
+    - None
+- `MANAGE_REPORT_PROCESSOR_ROLE`
+    - None
+- `address[] _memberAddresses`
+    - Current LidoOracle Committee (migrated in UpgradeTemplate)
+
+### **OracleDaemonConfig**
+*OZ, No Proxy*
+
+- `DEFAULT_ADMIN_ROLE`
+    - Agent
+- `CONFIG_MANAGER_ROLE`
+    - None
+
+### **OracleReportSanityChecker**
+*OZ, No Proxy*
+
+- `DEFAULT_ADMIN_ROLE` (set in constructor)
+    - Agent
+- `ALL_LIMITS_MANAGER_ROLE`
+- `CHURN_VALIDATORS_PER_DAY_LIMIT_MANAGER_ROLE`
+- `ONE_OFF_CL_BALANCE_DECREASE_LIMIT_MANAGER_ROLE`
+- `ANNUAL_BALANCE_INCREASE_LIMIT_MANAGER_ROLE`
+- `SHARE_RATE_DEVIATION_LIMIT_MANAGER_ROLE`
+- `MAX_VALIDATOR_EXIT_REQUESTS_PER_REPORT_ROLE`
+- `MAX_ACCOUNTING_EXTRA_DATA_LIST_ITEMS_COUNT_ROLE`
+- `MAX_NODE_OPERATORS_PER_EXTRA_DATA_ITEM_COUNT_ROLE`
+- `REQUEST_TIMESTAMP_MARGIN_MANAGER_ROLE`
+- `MAX_POSITIVE_TOKEN_REBASE_MANAGER_ROLE`
+    - None for all roles listed above
+
+### **ValidatorExitBusOracle (OZ)**
+*OZ, Proxy*
+
+- PROXY ADMIN
+    - Agent
+- `DEFAULT_ADMIN_ROLE`
+    - Agent
+- `PAUSE_ROLE` (in initializer)
+    - Gate Seal
```

Comment:
don't need Agent as it will grant the role to self first, right?

Reply:
yep